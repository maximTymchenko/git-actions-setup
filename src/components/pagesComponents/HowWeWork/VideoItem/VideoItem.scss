@import '../../../../styles/vars';

.video-item {
    &__container {
        cursor: pointer;
        width: 355px;
        height: 100%;
        max-height: 355px;
        margin-top: 40px;
        display: flex;
        flex-direction: column;
        outline: 2px solid $secondary;
        box-shadow: 7px 7px 1px 1px hsla(199, 58%, 51%, 0.4);
        @include transition;
        background-color: #fff;

        @media screen and (max-width: $medium__desktop) {
            margin-right: 80px;
        }

        @media screen and (max-width: $small__tablet) {
            width: 242px;
            height: 242px;
            position: relative;
        }

        //! test small container
        &--medium {
            width: 300px;
            height: 300px;
            justify-content: flex-start;
        }

        &--small {
            margin-right: 30px;
            width: 290px;
            height: 100%;

            @media screen and (max-width: $small__tablet) {
                width: 242px;
                height: 242px;
                position: relative;
            }
        }

        &:hover {
            outline: 4px solid $secondary;
            outline-offset: -2px;
        }

        &:active {
            @include button__press;
            box-shadow: 7px 7px 1px 1px hsla(199, 58%, 51%, 0);
        }
    }

    &__button {
        &--center {
            display: flex;
            justify-content: center;
            align-items: center;
        }

        opacity: 1;
        @include transition;

        @media screen and (max-width: $small__tablet) {
            //     position: absolute;
            //     top: 60px;
            //     right: 45px;
        }
    }

    &__number {
        font-family: 'OktaNeue-Heavy';
        font-size: 66px;
        font-weight: 900;
        line-height: 1.36;
        color: $secondary;
        padding: 5px 25px;

        @media screen and (max-width: $small__tablet) {
            font-size: 38px;
        }
    }

    &__label {
        font-family: 'OktaNeue-Heavy';
        font-size: 25px;
        padding: 0px 25px;
        max-width: 200px;
        font-weight: 900;
        color: $secondary;

        &--smallest {
            //! test small container
            font-size: 22px;
            padding-left: 10px;
            max-width: 400px;
        }

        @media screen and (max-width: $small__tablet) {
            font-size: 19px;
            position: absolute;
            bottom: 0;
        }
    }
}

.video-item__container:hover .video-item__button {
    opacity: 1;
}
