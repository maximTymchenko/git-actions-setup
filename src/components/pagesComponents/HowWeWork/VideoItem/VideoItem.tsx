import React, { FC } from 'react'
import './VideoItem.scss'
import Button from '../../../common/Button'

interface IVideo {
  num: string
  label: string
  buttonName: string
  buttonDesc: string
  buttonIcon: string
  size: string
  buttonType: string
  labelType: string
  position: string
  numberType: string
}

const VideoItem: FC = (props: IVideo) => {
  return (
    <div data-testid="video-item" className="video-item">
      <div
        className={`video-item__container video-item__container--${props.size}`}
      >
        <span
          className={`video-item__number video-item__number--${props.numberType}`}
        >
          {props.num}
        </span>
        <div
          className={`video-item__button video-item__button--${props.position}`}
        >
          <Button
            type={props.buttonType}
            className="button"
            children={props.buttonName}
            desc={props.buttonDesc}
            icon={props.buttonIcon}
          />
        </div>
        <span
          className={`video-item__label video-item__label--${props.labelType}`}
        >
          {props.label}
        </span>
      </div>
    </div>
  )
}

export default VideoItem
