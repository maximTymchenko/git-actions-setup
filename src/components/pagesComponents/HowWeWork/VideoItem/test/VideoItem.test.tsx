import React from 'react'
import VideoItem from '../VideoItem'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'

it('renders elem correctly', () => {
  const { getByTestId } = render(
    <VideoItem size="small" label="test" num="1"></VideoItem>
  )
  expect(getByTestId('video-item')).toBeTruthy()
})

it('renders without unexpected changes', () => {
  const tree = renderer
    .create(<VideoItem size="small" label="test" num="1"></VideoItem>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
