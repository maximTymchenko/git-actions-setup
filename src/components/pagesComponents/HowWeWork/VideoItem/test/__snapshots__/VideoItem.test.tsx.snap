// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without unexpected changes 1`] = `
<div
  className="video-item"
  data-testid="video-item"
>
  <div
    className="video-item__container video-item__container--small"
  >
    <span
      className="video-item__number"
    >
      1
    </span>
    <div
      className="video-item__button"
    >
      <button
        className="button button--undefined button--undefined button--undefined"
        data-testid="button"
      >
        <img
          className="button__icon"
          data-testid="button__icon"
        />
      </button>
    </div>
    <span
      className="video-item__label"
    >
      test
    </span>
  </div>
</div>
`;
